### PR TITLE
Fix freeze at wow.exe ( not responding and up CPU 100%)

### PR DESCRIPTION
**Pull request by alex1kiss**
Is work at me :)

### CODE REVIEW

**Comment by ccrs** [comment]
steps to reproduce the bug?

File: src/server/game/Entities/Creature/Creature.cpp [hunk 1]
```diff
@@ -1533,7 +1533,7 @@ float Creature::GetAttackDistance(Unit const* player) const
 void Creature::setDeathState(DeathState s)
 {
     Unit::setDeathState(s);
-
```
Comment (tkrokli) [inline src/server/game/Entities/Creature/Creature.cpp]: Please remove this indent.

File: src/server/game/Entities/Creature/Creature.cpp [hunk 2]
```diff
@@ -1565,6 +1566,7 @@ void Creature::setDeathState(DeathState s)
 
         Unit::setDeathState(CORPSE);
     }
+
```
Comment (tkrokli) [inline src/server/game/Entities/Creature/Creature.cpp]: This indent is also unneeded.

**Comment by dansanti** [comment]
@ccrs Fix this bug

https://github.com/TrinityCore/TrinityCore/issues/15263

**Comment by NorthStrider** [comment]
happens due to dirty database stuff. Emotes such as melee attacks that are added dbside in the addon tables will cause units to crash the client if they die. This is not handled properly. emotes are handled via timed events or animKits.

**Comment by tkrokli** [comment]
Thanks for the reminder. I had almost forgotten the wolves in Nortshire Valley. :-p

**Comment by alex1kiss** [comment]
Yes, it works with fix by tkrokli in DB :+1: 

DELETE FROM `creature_addon` WHERE `guid` IN (279887,279888,280006,280008,280004,280002,280000,279999,279999,280003,280001,279945,279928,280020,279891, 280016, 280019, 280013, 279998, 279956, 279947, 279938, 279929, 279914, 279916, 279906, 279905, 279894, 279889, 279893);

sorry for the distraction but I wanted to help with this fix ... because I am irritated by "Crush" . Thanks a lot :)

**Comment by tkrokli** [comment]
Thanks for the vote of confidence, but like I said in #15263, it is not a permanent fix.
Regardless, I updated + sorted the query (found an unnecessary duplicate GUID in the old one):
```sql
DELETE FROM `creature_addon` WHERE `guid` IN (279887,279888,279889,279891,279893,279894,279905,279906,279914,279916,279928,279929,279938,279945,279947,279956,279998,279999,280000,280001,280002,280003,280004,280006,280008,280013,280016,280019,280020);
```
Don't know if this PR should be closed already, the fix from @Melvin1263 looks like a better workaround.
(I still think something should be fixed in the way emotes are handled, but that will take more work.)

This is the only actual change in the PR, it would be nice if you could reopen:
```diff
diff --git a/src/server/game/Entities/Creature/Creature.cpp b/src/server/game/Entities/Creature/Creature.cpp
index 5c9bb01..45efe66 100644
--- a/src/server/game/Entities/Creature/Creature.cpp
+++ b/src/server/game/Entities/Creature/Creature.cpp
@@ -1547,6 +1547,7 @@ void Creature::setDeathState(DeathState s)
         SetUInt64Value(UNIT_NPC_FLAGS, UNIT_NPC_FLAG_NONE);
 
         SetUInt32Value(UNIT_FIELD_MOUNTDISPLAYID, 0); // if creature is mounted on a virtual mount, remove it at death
+        SetUInt32Value(UNIT_NPC_EMOTESTATE, 0); // fix client crash
 
         setActive(false);
 
```

**Pull request closed without merging**